src/org/jitsi/impl/neomedia/rtp/translator/SimulcastController.java

```diff
@@ -274,17 +281,34 @@ public void setTargetIndex(int newTargetIdx)
                 return;
             }
 
+            RTPEncodingDesc[] sourceEncodings = sourceTrack.getRTPEncodings();
+
             int currentTL0Idx = bitstreamController.getCurrentIndex();
             if (currentTL0Idx > -1)
             {
-                currentTL0Idx = sourceTrack.getRTPEncodings()
-                    [currentTL0Idx].getBaseLayer().getIndex();
+                currentTL0Idx
+                    = sourceEncodings[currentTL0Idx].getBaseLayer().getIndex();
             }
 
             int targetTL0Idx
-                = sourceTrack.getRTPEncodings()[newTargetIdx].getBaseLayer().getIndex();
+                = sourceEncodings[newTargetIdx].getBaseLayer().getIndex();
 
-            if (currentTL0Idx != targetTL0Idx)
+            // Something lower than the current must be streaming.
+            boolean sendFIR = targetTL0Idx < currentTL0Idx;
```

Comment:
That's a good suggestion but if `targetTL0Idx` is inactive, we might still have a chance with some other `idx` that satisfies `currentTL0Idx < idx < targetTL0Idx`. For example, if currentTL0Idx refers to 180p@3.5fps and targetTL0Idx refers to 720p@30fps, and targetTL0Idx is inactive, but 360p@30fps is streaming, we still want to ask for a FIR.